### PR TITLE
Add `vrf_address_family` module

### PR DESCRIPTION
**Pull request by Ruchip16**
##### SUMMARY

This PR adds support for `iosxr_vrf_address_family` resource module

##### DESCRIPTION

 Here is the [model](https://github.com/ansible-network/resource_module_models/pull/243) design for the `iosxr_vrf_address_family` resource module

 

##### ISSUE TYPE

- Feature Pull Request
- New Module Pull Request

##### COMPONENT NAME

`cisco.iosxr.iosxr_vrf_address_family`

##### MILESTONES

- [x] Parsing entire argspec ( which covers gathered and parsed state )
- [x] Merged and Deleted state working
- [x] Replaced state working
- [x] Overridden state working
- [x]  Unit tests & Integration tests

### CODE REVIEW

File: tests/integration/targets/iosxr_vrf_address_family/tests/common/_remove_config.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+- name: Remove VRF global configurations
+  cisco.iosxr.iosxr_config:
+    lines:
+      - no vrf VRF4
+      - no vrf VRF6
+      - no vrf VRF7
+      - no vrf test
+  register: result
```
Comment (roverflow) [inline tests/integration/targets/iosxr_vrf_address_family/tests/common/_remove_config.yaml]: might have to add a new line here

File: tests/integration/targets/iosxr_vrf_address_family/tests/common/deleted.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    - ansible.builtin.assert:
+        that:
+          - result.changed == true
+
+    - name: Assert that correct set of commands were generated
+      ansible.builtin.assert:
+        that:
+          - "{{ deleted['commands'] == result['commands'] }}"
+
+    # - name: Assert that before dicts are correctly generated
```
Comment (roverflow) [inline tests/integration/targets/iosxr_vrf_address_family/tests/common/deleted.yaml:24]: might have to remove this commented code if it is not going to be used

File: README.md [hunk 3]
```diff
@@ -69,7 +69,9 @@ Name | Description
 [cisco.iosxr.iosxr_static_routes](https://github.com/ansible-collections/cisco.iosxr/blob/main/docs/cisco.iosxr.iosxr_static_routes_module.rst)|Resource module to configure static routes.
 [cisco.iosxr.iosxr_system](https://github.com/ansible-collections/cisco.iosxr/blob/main/docs/cisco.iosxr.iosxr_system_module.rst)|Module to manage the system attributes.
 [cisco.iosxr.iosxr_user](https://github.com/ansible-collections/cisco.iosxr/blob/main/docs/cisco.iosxr.iosxr_user_module.rst)|Module to manage the aggregates of local users.
+[cisco.iosxr.iosxr_vrf_address_family](https://github.com/ansible-collections/cisco.iosxr/blob/main/docs/cisco.iosxr.iosxr_vrf_address_family_module.rst)|Resource module to configure VRF Address family.
 [cisco.iosxr.iosxr_vrf_global](https://github.com/ansible-collections/cisco.iosxr/blob/main/docs/cisco.iosxr.iosxr_vrf_global_module.rst)|Manages global VRF configuration.
+[cisco.iosxr.iosxr_vrfs](https://github.com/ansible-collections/cisco.iosxr/blob/main/docs/cisco.iosxr.iosxr_vrfs_module.rst)|Manages global VRF configuration.
```
Comment (roverflow) [inline README.md:74]: should this be here

File: plugins/module_utils/network/iosxr/facts/vrf_address_family/vrf_address_family.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        ansible_facts["ansible_network_resources"].update(facts)
+
+        return ansible_facts
+
+    def _post_parse(self, af_data):
+        """Converts the intermediate data structure
+            to valid format as per argspec.
+        :param obj: dict
+        """
+        af = af_data.get("address_families", {})
```
[the remark this answers is not in the document]
Comment (Ruchip16) [reply plugins/module_utils/network/iosxr/facts/vrf_address_family/vrf_address_family.py:101]: no i am converting the dict to  list of dict here

File: plugins/modules/iosxr_vrf_address_family.py [hunk 5]
```diff
@@ -0,0 +1,856 @@
...
+                vrf:
+                  advertise_as_vpn: true
+            maximum:
+              prefix: 100
+    state: merged
+
+# Task Output:
+# ------------
+#
+# before: {}
```
Comment (rohitthakur2590) [inline plugins/modules/iosxr_vrf_address_family.py:183]: why `before` is a dict here?

**Review by rohitthakur2590 — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -69,6 +69,7 @@ Name | Description
 [cisco.iosxr.iosxr_static_routes](https://github.com/ansible-collections/cisco.iosxr/blob/main/docs/cisco.iosxr.iosxr_static_routes_module.rst)|Resource module to configure static routes.
 [cisco.iosxr.iosxr_system](https://github.com/ansible-collections/cisco.iosxr/blob/main/docs/cisco.iosxr.iosxr_system_module.rst)|Module to manage the system attributes.
 [cisco.iosxr.iosxr_user](https://github.com/ansible-collections/cisco.iosxr/blob/main/docs/cisco.iosxr.iosxr_user_module.rst)|Module to manage the aggregates of local users.
+[cisco.iosxr.iosxr_vrf_address_family](https://github.com/ansible-collections/cisco.iosxr/blob/main/docs/cisco.iosxr.iosxr_vrf_address_family_module.rst)|Resource module to configure VRF Address family.
```
Comment (rohitthakur2590) [inline README.md:72]: Please add the redirect for the RM in meta/runtime/.yml file

**Review by rohitthakur2590 — CHANGES_REQUESTED** [review]

**Review by rohitthakur2590 — APPROVED** [review]

File: changelogs/fragments/add_vrfs_module.yaml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+---
+minor_changes:
+  - Adds a new module `iosxr_vrfs` to manage VRFs on Cisco IOS-XR devices.
```
Comment (KB-perByte) [inline changelogs/fragments/add_vrfs_module.yaml:3]: Can we please make it just one changelog entry, we need not add 3 list entries for one module that is being added.

File: plugins/modules/iosxr_vrf_address_family.py [hunk 8]
```diff
@@ -0,0 +1,856 @@
...
+"""
+
+EXAMPLES = """
+
+# Using merged
+#
+# Before state:
+# -------------
+#
+# RP/0/0/CPU0:iosxr-02#show running-config vrf
```
Comment (KB-perByte) [inline plugins/modules/iosxr_vrf_address_family.py:145]: Simplify the hostname, looks cleaner and remove the timestamp please

File: plugins/modules/iosxr_vrf_address_family.py [hunk 9]
```diff
@@ -0,0 +1,856 @@
...
+# Fri Feb  9 07:02:35.789 UTC
```
Comment (KB-perByte) [inline plugins/modules/iosxr_vrf_address_family.py:146]: ^ the timestamp can be removed -- applies to all the examples below

**Review by KB-perByte — CHANGES_REQUESTED** [review]

**Review by KB-perByte — APPROVED** [review]

**Pull request merged**